2024/YoctoProjectDeveloperDay/practical-sec-tee.md

```diff
@@ -0,0 +1,994 @@
+
+extras:
+- What is Trusted Execution Environment?
+  + Normal World vs Secure World
+    - Different architecture overview
+  + Root of trust
+  + TEE use cases
+    - Secure storage
+
+- TEE practice (Daniil Klimuk)
+    - Introduction
+    - What to integrate and why?
+    - Compilation from source vs vendor binaries.
+    - Linux/U-Boot and Yocto integration.
+    - OP-TEE Secure Storage.
+    - Experiments on Rockchip RK3566 SoC.
+    - Zarhus OS.
+- Outro and sources.
+
+???
+
+- Time for this slide: 30s
+- Idea/goal of this slide: Present entire presentation in a short way
+- What to say:
+    - TODO
+- Notes:
+    - TODO
+
+---
+
+# What is Trusted Execution Environment?
+
+.center[ <img src="/img/TEE_High_Lvl_Overview.svg" height="250px"> ]
+
+<br>
+
+_"A trusted execution environment (TEE) means a secure area which can guarantee
+the confidentiality and integrity of the code and data inside of this area._
+
+???
+
+- "from general information to specifics"
+- There's no agreed on firm definition and a lot of sources define their own
+- Definition taken from _"Building Secure Firmware, Armoring the Foundation of
+the Platform"_
+  + Will be listed in resources
+  + **Usually** refers to the possibility of TEE being only isolated via
+software which is not the case in some definitions.
+- Can be isolated via software
+- The term TEE was coined somewhere in 2000s
+- Used to be "secure area of the main processor" but the definition has
+broadened
+  + Can even be a separate device, more on that later
+- A device can have multiple TEEs nowadays
+- Used implementation is mostly dependant on underlying CPU architecture
+
+---
+
+# Common Ground in TEE definition
+
+### What it ensures:
+
+- **Data integrity**: preventing unauthorized entities from altering data when
+data is being processed.
+
+- **Data confidentiality**: Guarantees the authenticity of the executed code.
+
+- **Code integrity**: The confidentiality of its code, data and runtime states
+stored on a persistent memory.
+
+???
+
+- This is the part that is universally agreed on
+- This exact definition is from Confidential Computing Consortium
+  - A Linux Foundation Project which aims to accelerate the adoption of Trusted
+  Execution Environment (TEE) technologies and standards
+  - With Confidential Computing being the protection of data in use by
+  performing computation in a hardware-based, attested Trusted Execution
+  Environment.
+
+---
+
+# TEE locations
+
+.right-column45[
+.center[
+
+**CPU based TEE**
+
+<img src="/img/TEE_CPU_based.svg" width="250px">
+
+]
+]
+
+.left-column70[
+### Examples
+
+- Arm Trustzone
+- Intel SGX (Software Guard Extensions)
+- AMD SEV (Secure Encrypted Virtualization)
+
+]
+
+---
+
+# TEE locations
+
+.right-column45[
+.center[
+
+**Coprocessor Based TEE (One SoC)**
+
+<img src="/img/TEE_CoCPU_based.svg" width="250px">
+
+]
+]
+
+.left-column70[
+### Examples
+
+- Integrated TPM (Trusted Platform Module)
```

Comment:
Is TPM a TEE? It is rather a part of it.

Reply:
Depends who you ask to be honest. Some sources cite it as a TEE and it fits the definition that I used in previous slides. Building Secure Firmware specifies TPM as a type of Co-processor based TEE (Chapter 17/Co-processor based TEE). It could also as a static hardware-based root of trust that extends the chain of trust to TEE

Reply:
I see, not critical for now then.